### PR TITLE
[docs] Update readme

### PR DESCRIPTION
**Pull request by stevhliu**
This PR updates the `README` to only include a brief intro to the library and adds how to navigate the docs to find what you're looking for and a table for all the supported pipelines.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,536 +15,147 @@
...
-prompt_ids, processed_masked_images, processed_masks = pipeline.prepare_inputs(prompt, init_image, mask_image)
-
-
-# shard inputs and rng
-params = replicate(params)
-prng_seed = jax.random.split(prng_seed, jax.device_count())
-prompt_ids = shard(prompt_ids)
-processed_masked_images = shard(processed_masked_images)
-processed_masks = shard(processed_masks)
+from diffusers import DiffusionPipeline
```
Comment (yiyixuxu) [inline README.md:61]: Maybe here we should use the `StableDiffusionPipeline` example with `fp16`  on the current readme page?  

```python
import torch
from diffusers import StableDiffusionPipeline

pipe = StableDiffusionPipeline.from_pretrained("runwayml/stable-diffusion-v1-5", torch_dtype=torch.float16)
pipe = pipe.to("cuda")

prompt = "a photo of an astronaut riding a horse on mars"
image = pipe(prompt).images[0]  
```

File: README.md [hunk 2]
```diff
@@ -15,536 +15,147 @@
...
-images = pipeline.numpy_to_pil(np.asarray(images.reshape((num_samples,) + images.shape[-3:])))
+pipeline = DiffusionPipeline.from_pretrained("runwayml/stable-diffusion-v1-5")
+pipeline.to("cuda")
+pipeline("An image of a squirrel in Picasso style").images[0]
 ```
 
-### Image-to-Image text-guided generation with Stable Diffusion
-
-The `StableDiffusionImg2ImgPipeline` lets you pass a text prompt and an initial image to condition the generation of new images.
+You can also dig into the models and schedulers toolbox to build your own diffusion system:
```
Comment (yiyixuxu) [inline README.md:68]: I don't think this example needs to be here - I think maybe we can put a link instead

Maybe instead, we can show an example of how to swap out individual components since that's a more common way to use diffusers as a toolbox.

wait for my teammates to chime in

File: README.md [hunk 3]
```diff
@@ -15,536 +15,147 @@
...
+previous_noisy_sample = scheduler.step(noisy_residual, t, input).prev_sample
+input = previous_noisy_sample
+
+image = (input / 2 + 0.5).clamp(0, 1)
+image = image.cpu().permute(0, 2, 3, 1).numpy()[0]
+image = Image.fromarray((image * 255)).round().astype("uint8")
+image
+```
+
+Check out the [Quickstart](https://huggingface.co/docs/diffusers/quicktour) to launch your !
```
Comment (yiyixuxu) [inline README.md:97]: launch your ?
think this sentence is missing a word 😂

File: README.md [hunk 4]
```diff
@@ -15,536 +15,147 @@
...
+
+## How to navigate the documentation
+
+| **Documentation**                                                   | **What can I learn?**                                                                                                                                                                           |
+|---------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Tutorial                                                            | A basic crash course for learning how to use the library's most important features like using models and schedulers to build your own diffusion system, and training your own diffusion model.  |
```
Comment (yiyixuxu) [inline README.md:103]: we should add links to the doc for these sections?

**Review by yiyixuxu — APPROVED** [review]
left some comments, but overall I can't believe how much it improved our current README (it is currently super cluttered and overwhelming with information) - Thank you so much! 

@williamberman @pcuenca @sayakpaul,  can you all take a look? This page is super important

File: README.md [hunk 5]
```diff
@@ -15,536 +15,147 @@
     </a>
 </p>
 
-🤗 Diffusers provides pretrained diffusion models across multiple modalities, such as vision and audio, and serves
-as a modular toolbox for inference and training of diffusion models.
+🤗 Diffusers is the go-to library for state-of-the-art pretrained diffusion models for generating images, audio, and even 3D structures of molecules. Whether you're looking for a simple inference solution or training your own diffusion models, 🤗 Diffusers is a modular toolbox that supports both. Our library is designed with a focus on [usability over performance](https://huggingface.co/docs/diffusers/conceptual/philosophy#usability-over-performance), [simple over easy](https://huggingface.co/docs/diffusers/conceptual/philosophy#simple-over-easy), and [customizability over abstractions](https://huggingface.co/docs/diffusers/conceptual/philosophy#tweakable-contributorfriendly-over-abstraction).
```
Comment (patrickvonplaten) [inline README.md:18]: Nice

File: README.md (hunk 1 above)
Comment (patrickvonplaten) [reply README.md:61]: Actually I like that we import `DiffusionPipeline` here instead of `StableDiffusionPipeline` => we should try to nudge the community a bit towards using the `DiffusionPipeline` class only.

Comment (patrickvonplaten) [reply README.md:61]: I'd maybe only add a `torch_dtype=torch.float16` in the beginnin as it gives a big speed-up

File: README.md [hunk 6]
```diff
@@ -15,536 +15,147 @@
...
+|---|---|:---:|:---:|
+| [alt_diffusion](./api/pipelines/alt_diffusion) | [**AltDiffusion**](https://arxiv.org/abs/2211.06679) | Image-to-Image Text-Guided Generation |
+| [audio_diffusion](./api/pipelines/audio_diffusion) | [**Audio Diffusion**](https://github.com/teticio/audio-diffusion.git) | Unconditional Audio Generation | [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/teticio/audio-diffusion/blob/master/notebooks/audio_diffusion_pipeline.ipynb)
+| [controlnet](./api/pipelines/stable_diffusion/controlnet) | [**ControlNet with Stable Diffusion**](https://arxiv.org/abs/2302.05543) | Image-to-Image Text-Guided Generation | [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/huggingface/notebooks/blob/main/diffusers/controlnet.ipynb)
+| [cycle_diffusion](./api/pipelines/cycle_diffusion) | [**Cycle Diffusion**](https://arxiv.org/abs/2210.05559) | Image-to-Image Text-Guided Generation |
+| [dance_diffusion](./api/pipelines/dance_diffusion) | [**Dance Diffusion**](https://github.com/williamberman/diffusers.git) | Unconditional Audio Generation |
```
[the remark this answers is not in the document]
Comment (patrickvonplaten) [reply README.md:117]: Can we here maybe remove the "Colab" column as those colabs are not maintainable ?

**Review by patrickvonplaten — APPROVED** [review]
Looks great! Just left some nits

File: README.md (hunk 3 above)
Comment (stevhliu) [reply README.md:97]: Haha oops 🥲

File: README.md (hunk 4 above)
Comment (stevhliu) [reply README.md:103]: Only the Training section has an overview page that can be linked to. We can add a link to the first page of each of the other sections (Tutorial, Loading, Pipelines, Optimization), but that would just take you to a doc about a specific topic instead of describing more generally what you learn in this section. 

The other option would be to create an Overview page for each section like Training does, which I think would be better 🙂

**Comment by patrickvonplaten** [comment]
Very cool! Feel free to merge whenever @stevhliu :-)
